src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,153 @@
 package core.basesyntax;
 
+//import java.util.HashMap;
+import java.util.Objects;
+import java.util.Random;
+
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public double EXPANSION_COEFFICIENT = 2;
+    public double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] values;
+    private static final int defaultCapacity = 16;
+    private int size;
+    private Integer threshold;
+
+    public MyHashMap() {
+        values = new Node[defaultCapacity];
+        this.threshold = (int) (defaultCapacity * LOAD_FACTOR);
+        size = 0;
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> nextNode;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public void setKey(K key) {
+            this.key = key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public Node<K, V> getNextNode() {
+            return nextNode;
+        }
+
+        public void setNextNode(Node<K, V> nextNode) {
+            this.nextNode = nextNode;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            Node<K, V>[] oldBucketStock = values;
+            values = new Node[values.length * 2];
+            for (int i = 0; i < oldBucketStock.length; i++) {
+                while (oldBucketStock[i] != null) {
+                    K k = oldBucketStock[i].key;
```

Comment:
Please avoid one-character names even if it is of type K or V. You can name this like `key` and `value` appropriately.